### DISCUSSION TITLE
what is the memory cost for Lunatic's super lightweight processes

### DISCUSSION THREAD

**Discussion by superdolt** [lunatic-solutions/lunatic#64] [Q&A]
how many kb / bytes per process?

**Comment by bkolobara** [comment]
It's a bit hard to answer this because it really depends on many factors. Like what the languages is that is compiled to wasm, how many wasm memories and tables are used. lunatic also assumes that virtual memory is "free" on modern x64 processors and we over-allocate virtual memory a lot to avoid bound checks on memory access.

The easiest way of measuring this would be just to spawn 100k processes from a particular wasm module you are interested in and measure memory usage before and after.

**Reply by superdolt** [reply]
do u hv a rough idea on how much memory per connection is using?

**Reply by bkolobara** [reply]
I don't have a number, but I don't think this would be a bottleneck for any real-world app.

**Reply by superdolt** [reply]
so what's the max simultaneous connection we can have per server using total memory size of...?
